distributed/diagnostics/tests/test_scheduler_plugin.py

```diff
@@ -115,6 +117,96 @@ async def remove_worker(self, worker, scheduler):
     assert events == []
 
 
+@gen_cluster(nthreads=[])
+async def test_async_and_sync_remove_worker(s):
+    events = []
+
+    class MyAsyncPlugin(SchedulerPlugin):
+        name = "MyAsyncPlugin"
+
+        def __init__(self) -> None:
+            super().__init__()
+            self.in_remove_worker = asyncio.Event()
+            self.block_remove_worker = asyncio.Event()
+
+        async def remove_worker(self, worker, scheduler):
+            assert scheduler is s
+            self.in_remove_worker.set()
+            await self.block_remove_worker.wait()
+            events.append(("async_remove_worker", worker))
+
+    class MySyncPlugin(SchedulerPlugin):
+        name = "MySyncPlugin"
+
+        def remove_worker(self, worker, scheduler):
+            assert scheduler is s
+            events.append(("sync_remove_worker", worker))
```

Comment:
Do we have access to the name you are providing to `Scheduler.add_plugin`? If so, this would make the test cases below easier to read

Reply:
Unfortunately, we do not have that (yet).

Reply:
I've rewritten the code a bit to make it (hopefully) easier to read and add delayed `add_worker`.